php/class-plugin.php

```diff
@@ -313,7 +313,7 @@ function( $post ) use ( $is_multiple_post_types, $include_featured_images ) {
 					'post_author' => $post->post_author,
 				);
 				if ( $include_featured_images ) {
-					$attachment_id = get_post_thumbnail_id( $post->ID );
+					$attachment_id = get_post_meta( $post->ID, $include_featured_images, true ) ?: get_post_thumbnail_id( $post->ID );
```

Comment:
@lgedeon I suggest changing the approach. Instead of overloading this query parameter to allow the client to supply the postmeta key for the featured media, you can instead introduce a filter for this. So I suggest you revert all of the changes here and instead replace this and introduce a new filter like this:

```php
$attachment_id = get_post_thumbnail_id( $post->ID );

/**
 * Filters the attachment ID for the featured image for a given post.
 *
 * @param int|bool $attachment_id Attachment ID or `false` if no featured image is associated.
 * @param \WP_Post $post Post.
 */
$attachment_id = apply_filters( 'customize_object_selector_featured_image', $attachment_id, $post );
```

This will then allow you to add a filter—on the server—in your the custom post type definition like so:

```php
add_filter( 'customize_object_selector_featured_image', function( $attachment_id, $post ) {
    if ( 'staff' === $post->post_type ) {
        $attachment_id = get_post_meta( $post->ID, 'head_shot', true );
    }
    return $attachment_id;
}, 10, 2 );
```